nominal/core/dataset_file.py

```diff
@@ -36,8 +50,69 @@ def _from_conjure(cls, dataset_file: scout_catalog.DatasetFile) -> Self:
             uploaded_at=upload_time,
             ingested_at=ingest_time,
             ingest_status=IngestStatus._from_conjure(dataset_file.ingest_status),
+            _clients=clients,
         )
 
+    def download(self, destination: pathlib.Path) -> pathlib.Path:
+        """Download the dataset file to a destination on local disk.
+
+        Args:
+            destination: If an existing directory, downloads the file to the given directory.
+                Otherwise, downloads the file to the given path.
+
+        Returns:
+            Path that the file was downloaded to
+
+        Raises:
+            FileExistsError: File already exists at destination
+            RuntimeError: Error downloading file
+        """
+        uri = self._clients.catalog.get_dataset_file_uri(self._clients.auth_header, self.dataset_rid, self.id).uri
+        logger.info("Downloading %s (%s) => %s", self.name, uri, destination)
+        return download_presigned_uri(uri, destination)
+
+    def download_origin_files(self, destination: pathlib.Path) -> Sequence[pathlib.Path]:
+        """Download the origin files for a given dataset file to a destination on local disk.
+
+        Args:
+            destination: Destination to download file(s) to.
+                NOTE: If multiple files are requested for export, must be a directory.
+                NOTE: If a single file is requested for export, mest be an existing directory or the path
+                    to write the file as
+
+        Returns:
+            List of downloaded file locations.
+
+        Raises:
+            FileExistsError: File already exists at the destination
+        """
+        origin_uris = self._clients.catalog.get_origin_file_uris(self._clients.auth_header, self.dataset_rid, self.id)
+        if not origin_uris:
+            return []
+
+        if destination.exists():
+            if len(origin_uris) == 1 and destination.is_file():
+                raise FileExistsError(f"Cannot download origin file to {destination}: already exists!")
+            elif destination.is_file():
+                raise FileExistsError(f"Cannot download origin files to {destination}: already exists as a file!")
+        elif len(origin_uris) == 1:
+            destination.parent.mkdir(parents=True, exist_ok=True)
+        else:
+            destination.mkdir(parents=True, exist_ok=True)
```

Comment:
I don't understand why you would create the parent but not the dir itself if origin_uris is 1